lib/room.h

```diff
@@ -191,6 +191,7 @@ class Room : public QObject {
     QStringList altAliases() const;
     QStringList aliases() const;
     QString displayName() const;
+    QList<const RoomEvent*> pinnedEvents() const;
```

Comment:
There are two problems here. First, this introduces asymmetric API: the getter returns an array of events while the setter accepts a `QStringList`. Second, `QList<const RoomEvent*>` is a bit difficult to tackle from QML if you expose pinned events in a `Q_PROPERTY` (I recommend to do it). Perhaps it's worth having the following:
- add `pinnedEventsChanged()` signal; emitting that signal can be added in `processStateEvent()`;
- make a `Q_PROPERTY` for pinned event ids (with a setter, a getter, and a `NOTIFY` signal - see the previous point);
- keep the current `pinnedEvents()` accessor, with a comment that it only returns locally available events, and the full list should be taken from `pinnedEventIds()`; one thing that I'd change here is using `QVector` instead of `QList` since `QList` is a [pretty crappy container](https://marcmutz.wordpress.com/2010/07/29/sneak-preview-qlist-considered-harmful/).